sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ReadOnlyRequestOptions.java

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosDiagnosticsThresholds;
+import com.azure.cosmos.CosmosEndToEndOperationLatencyPolicyConfig;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * Getters for the common request options for operations in CosmosDB.
+ */
+public interface ReadOnlyRequestOptions {
+
+    /**
+     * Gets the CosmosEndToEndLatencyPolicyConfig.
+     *
+     * @return the CosmosEndToEndLatencyPolicyConfig.
+     */
+    default CosmosEndToEndOperationLatencyPolicyConfig getCosmosEndToEndLatencyPolicyConfig() {
+        LoggerFactory.getLogger(ReadOnlyRequestOptions.class).info("This is not the correct class to get the CosmosEndToEndLatencyPolicyConfig.");
```

Comment:
We can use a class-level logger right with the `static` and `final` qualifiers which should be allowed for an interface even.

Reply:
+1

Reply:
I initially did this but, I went against it because I didn't know the implications of making the logger public because an interface can't have private variables. Additionally, I did it to avoid creating more public api.

Reply:
Let's discuss this over a call. I don't quite get why we even have default implementations in the first place - i mean the only thing we log is that it is wrong to not have overridden the method - why not enforce that?

Reply:
Based on the call, moved the default implementation to all the classes. Called out in the docs that it would return null if tried for the wrong operation. Added todos for the ones that could be implemented in the future.